Shared/Samples/Show portal user info/ShowPortalUserInfoView.swift

```diff
@@ -0,0 +1,304 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct ShowPortalUserInfoView: View {
+    /// The data model that helps determine the view.
+    @State private var model = Model()
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        VStack {
+            // Shows field for custom portal urls as well as Sign In / Out and Load Portal functions.
+            portalDetails
+            // If loading, show that the user profile will display when complete.
+            if model.isLoading {
+                ContentUnavailableView(
+                    "Portal User Information",
+                    systemImage: "person.crop.circle.dashed",
+                    description: Text("Your portal user information will be displayed here.")
+                )
+                // Otherwise show the user information that was loaded.
+            } else {
+                InfoScreen(model: $model)
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+    
+    /// Asynchronously loads the portal user information.
+    private func loadUser() async {
+        do {
+            try await model.loadPortalUser()
+        } catch {
+            // If an error occurs, mark loading as true (to reset)
+            // and store the error to present an alert.
+            model.isLoading = true
+            self.error = error
+        }
+    }
+    
+    /// A composed view for handling portal login, logout, and URL input.
+    @ViewBuilder private var portalDetails: some View {
+        PortalDetailsView(
+            url: $model.portalURLString,
+            model: $model,
+            // Update the model when the portal URL is changed by the user.
+            onSetUrl: {
+                model.portalURLString = $0
+            },
+            // Sign out the user and reset state.
+            onSignOut: {
+                Task {
+                    await model.signOut()
+                }
+            },
+            // Load the user's portal data when sign in is triggered.
+            onLoadPortal: {
+                Task {
+                    await loadUser()
+                }
+            }
+        )
+        .frame(maxWidth: 400)
```

Comment:
You might actually consider moving this up the hierarchy, possibly onto the body's VStack. Otherwise, the dividers in InfoScreen will also grow to the full window width on Catalyst.